packages/embark-deploy-tracker/src/trackingFunctions.js

```diff
@@ -31,44 +34,56 @@ export default class TrackingFunctions {
       if (this._chains) {
         return this._chains;
       }
-      if (!this.enabled) {
+
+      const exists = await this.fs.exists(this.chainsFilePath);
+      if (!this.enabled || !exists) {
         this._chains = null;
         return this._chains;
       }
+
       this._chains = await this.fs.readJSON(this.chainsFilePath);
       return this._chains;
     })();
   }
 
+  set chains(chains) {
+    this._chains = chains;
+    // chains has changed, therefore currentChain should too
+    // reset the backing variable for currentChain so it can be recalculated on next get
+    this._currentChain = null;
+  }
+
   get currentChain() {
     return (async () => {
       if (this._currentChain) {
         return this._currentChain;
       }
+
       if (!this.enabled) {
-        this._currentChain = {contracts: []};
-        return this._currentChain;
+        return null;
       }
 
-      let block;
+      const chains = (await this.chains) || {};
+      const {hash} = await this.block;
+      this._currentChain = chains[hash];
+      return this._currentChain;
+    })();
+  }
+
+  get block() {
+    return (async () => {
+      if (this._block) {
+        return this._block;
+      }
       const web3 = await this.web3;
       try {
-        block = await web3.eth.getBlock(0, true);
+        this._block = await web3.eth.getBlock(0, true);
       } catch (err) {
         // Retry with block 1 (Block 0 fails with Ganache-cli using the --fork option)
-        block = await web3.eth.getBlock(1, true);
+        this._block = await web3.eth.getBlock(1, true);
```

Comment:
To be sure, what happens now if the second `getBlock` fails? I fixed this again on master but it seems like it got lost in a rebase: https://github.com/embark-framework/embark/pull/1801/files#diff-89ede650c6f0ddc7ebd3b3047601ed33R89

Basically, using --fork, there is a change that all `getBlock` calls fail, so we need to show an error, but not die.